libs/sdk-core/src/boltzswap.rs

```diff
@@ -77,18 +85,59 @@ struct Pairs {
     pairs: HashMap<String, Pair>,
 }
 
+pub struct BoltzApi {}
+
+#[tonic::async_trait]
+impl ReverseSwapperAPI for BoltzApi {
+    async fn reverse_swap_info(&self) -> Result<ReverseSwapInfo> {
+        crate::boltzswap::reverse_swap_info().await
+    }
+
+    async fn create_reverse_swap(
+        &self,
+        amount_sat: u64,
+        onchain_claim_address: String,
+        pair_hash: String,
+        routing_node: String,
+    ) -> Result<ReverseSwap> {
+        let rand_bytes: [u8; 32] = random();
+        let preimage = sha256::Hash::hash(&rand_bytes);
+        let preimage_hash = sha256::Hash::hash(&preimage);
+        let preimage_hash_hex = preimage_hash.to_hex();
+
+        let response_body = Client::new()
+            .post(CREATE_REVERSE_SWAP_ENDPOINT)
+            .header(CONTENT_TYPE, "application/json")
+            .body(get_boltz_reverse_swap_args(
+                amount_sat,
+                preimage_hash_hex,
+                pair_hash,
+                onchain_claim_address,
+                routing_node,
+            ))
+            .send()
+            .await?
+            .text()
```

Comment:
reqwest crate has a built in json feature where you can use the json() method to deserialize the response. Might be more elegant, see here: https://docs.rs/reqwest/latest/reqwest/struct.Response.html

Reply:
Good point, done.